lib/iris/_merge.py

```diff
@@ -1406,10 +1406,35 @@ def axis_and_name(name):
                     # TODO: Consider appropriate sort order (ascending,
                     # descending) i.e. use CF positive attribute.
                     cells = sorted(indexes[name])
-                    points = np.array(
-                        [cell.point for cell in cells],
-                        dtype=metadata[name].points_dtype,
-                    )
+                    points = [cell.point for cell in cells]
+
+                    # If any points are masked then create a masked array type,
+                    # otherwise create a standard ndarray.
+                    if np.ma.masked in points:
+                        dtype = metadata[name].points_dtype
+
+                        try:
+                            # This will always raise a numpy warning about "converting masked elements to Nan"
+                            # TODO: Can we pre-fill masked data in `points`? Would require
+                            #       generation of separate mask array...
+                            points = np.ma.masked_array(points, dtype=dtype)
+                        except np.ma.MaskError:
+                            # Fails for integer types as cannot convert a np.ma.masked to int type.
+                            # Need to loop over list and add points manually to a pre-masked array:
+
+                            # TODO: Perhaps use this method for all arrays, not just integer typea?
+                            # This will avoid numpy warnings about converting masked values to NaN
+                            # and also remove the need for this try..except block.
```

Comment:
Still in the works?

Reply:
Recent commit 5c9e553 uses it for both cases rather than the try/except block.